### PR TITLE
use PHP8's constructor property promotion

### PR DESCRIPTION
**Pull request by shdehnavi**
All class fields, constructor definition, and variable assignments are updated using constructor property promotion

### CODE REVIEW

File: lib/Controller/InternalController.php [hunk 1]
```diff
@@ -21,18 +21,16 @@
 
 class InternalController extends ApiController {
 	private $userId;
-	private $credentialService;
-	private $config;
 	private $manager;
```
Comment (binsky08) [inline lib/Controller/InternalController.php:24]: is there a reason leaving $manager, as it is?

File: lib/Controller/ShareController.php [hunk 2]
```diff
@@ -36,33 +36,26 @@
...
+		private IGroupManager $groupManager,
+		private IUserManager $userManager,
+		private ActivityService $activityService,
+		private VaultService $vaultService,
+		private ShareService $shareService,
+		private CredentialService $credentialService,
+		private NotificationService $notificationService,
+		private FileService $fileService,
+		SettingsService $config,
+		IManager $IManager,
```
Comment (binsky08) [inline lib/Controller/ShareController.php:58]: same here, is there a reason leaving SettingsService and IManager as it is?

File: lib/Migration/ServerSideEncryption.php [hunk 3]
```diff
@@ -39,35 +39,17 @@
...
-		$this->credentialService = $credentialService;
-		$this->revisionService = $revisionService;
-		$this->fileService = $fileService;
+	public function __construct(
+		private EncryptService $encryptService,
+		private IDBConnection $db,
+		private LoggerInterface $logger,
+		private CredentialService $credentialService,
+		private CredentialRevisionService $revisionService,
+		private FileService    $fileService, IConfig $config,
```
Comment (binsky08) [inline lib/Migration/ServerSideEncryption.php:51]: please fix the format with indentation and a new line [51]

File: lib/Service/NotificationService.php [hunk 4]
```diff
@@ -30,11 +30,12 @@
 class NotificationService {
 
 	private IManager $manager;
```
Comment (binsky08) [inline lib/Service/NotificationService.php:32]: same here

File: lib/Service/ShareService.php [hunk 5]
```diff
@@ -38,28 +38,18 @@
 use OCP\Notification\IManager;
 
 class ShareService {
-	private SharingACLMapper $sharingACL;
-	private ShareRequestMapper $shareRequest;
 	private CredentialMapper $credential;
```
Comment (binsky08) [inline lib/Service/ShareService.php:41]: same here

File: lib/Controller/InternalController.php (hunk 1 above)
Comment (shdehnavi) [reply lib/Controller/InternalController.php]: I wasn't sure whether it breaks the DI or not,  I've addressed it in the next commit

File: lib/Controller/ShareController.php (hunk 2 above)
Comment (shdehnavi) [reply lib/Controller/ShareController.php]: I wasn't sure whether it breaks the DI or not,  I've addressed it in the next commit

File: lib/Migration/ServerSideEncryption.php (hunk 3 above)
Comment (shdehnavi) [reply lib/Migration/ServerSideEncryption.php]: I've addressed it in the next commit

File: lib/Service/NotificationService.php (hunk 4 above)
Comment (shdehnavi) [reply lib/Service/NotificationService.php]: I've addressed it in the next commit

File: lib/Service/ShareService.php (hunk 5 above)
Comment (shdehnavi) [reply lib/Service/ShareService.php]: I've addressed it in the next commit

**Review by binsky08 — APPROVED** [review]
thanks for the pr :)

**Comment by shdehnavi** [comment]
@binsky08 The Scrutinizer shows a time-out warning, shall I do anything else?

**Comment by binsky08** [comment]
> @binsky08 The Scrutinizer shows a time-out warning, shall I do anything else?

The problem isn't you :D
it's just broken right now

**Review by animalillo — APPROVED** [review]

**Comment by animalillo** [comment]
Thanks for the pr! Code definitely looks better and now we don't need backwards compatibility with older php versions, so :heart:

**Pull request merged**